### PR TITLE
Re-factor cgm-rg

### PR DESCRIPTION
**Pull request by prajurock**
- Aggregation added as Result Generation  Class
- fastcore set_attr added

### CODE REVIEW

**Comment by prajurock** [comment]
Tests are failing because there are lots of change in class. Please suggest any suggestion which you all have to improve the code. After implementing your suggestion I will fix the tests.

**Review by markus-hinsche — COMMENTED** [review]
Looks great for me, because achieves to reduce code duplication. 
I do not understand why 2 tests have been removed

**Comment by prashant905** [comment]
@prajurock  we are experiencing some code deletion due to refactoring. We are trying to put that piece of code back into develop branch.  Please don't merge any refactor code into develop branch until the issue is resolved.

File: src/result_generation/result_generation.py
Comment (prajurock) [reply src/result_generation/result_generation.py:10]: Yes

**Review by markus-hinsche — APPROVED** [review]
LGTM

**Review by shanky0009 — APPROVED** [review]

**Comment by gangooteli** [comment]
This code is failing for RGBD models. Found it while reviewing and also verified it locally.

image

**Suggestions to Everyone:** 

- Please run it locally before submitting any code for review (atleast for such large changes).
- Try to make refactoring as small as possible in a single PR. In some of the previous PR also, saw some huge changes in the code in a single PR.
- In this refactoring, I am able to see changes in 17 files, which is hard to review and find mistakes. 
- It can only be found if gone through whole code. 
- Reviewers, Pls take your time to review the code.

**Review by gangooteli — CHANGES_REQUESTED** [review]
Added comment, pls fix issues of RGBD model before merge.

File: src/result_gen_with_api.py
Comment (gangooteli) [inline src/result_gen_with_api.py:396]: Make changes in line 300 also where RGBD class is being used

**Review by gangooteli — CHANGES_REQUESTED** [review]

**Pull request merged**
